### DISCUSSION TITLE
Feature Request : Fee Credit

### DISCUSSION THREAD

**Discussion by ghost** [stackernews/stacker.news#1190] [General]
Issue : 1188
     Title : Fee Credit
     Author : Radentor
     Status : Proposed
     Type : Feature
     Created : 2024-05-25

# Abstract
This issue describes a proposal to make the upcoming "fee credit" as the new default currency for zapping which later can be treated as tradeable token on Stacker News.

# Motivation

> It sounds like as a non-custodial receipient, I'm still going to be acquiring "fee credits" from non-wallet users, and I'm effectively unable to do a single thing with those besides upvote with them.

Quoted from SN comment [#523435](https://stacker.news/items/523435/r/Radentor?commentId=523843)

> Accumulating too many fee credits could be a waste of sats though since those are forever trapped inside SN... and I don't mean you don't deserve fees, but I probably spend 90% of sats on tips to other users at the moment.
> Do you plan to offer a market to trade fee credits?

Quoted from SN comment [#523435](https://stacker.news/items/523435/r/Radentor?commentId=523546)

> You better change the whole system of zaps: LITERALLY CONVERT THEM IN COWBOY TOKENS.
Make users to buy first these tokens, with real sats.
Then internally they can use it as zap tokens.
Yes sounds like a shitcoinery, but you can extend this to many things.
Like a gamification of SN. How you are playing World of Warcraft or something like that.
If they want to "sell" those tokens, make a SN internal auction and people can trade freely between them.
No more wallets, no more sats, no more trouble.

Quoted from SN comment [#523435](https://stacker.news/items/523435/r/Radentor?commentId=523531)

> We will trade cowboy tokens on Robosats
Or there could be a dedicated territory for offers and the trade could be done via DMs
> It would be preferred to be only inside SN, a special auction desk like in WoW game.

Quoted from SN comment [#523435](https://stacker.news/items/523435/r/Radentor?commentId=523643)

> This introduces a lot of friction to zapping UX and I dare say it will kill micro-zapping, if not zapping in general.
This means that people who intend to withdraw sats (instead of just stacking cowboy tokens), perhaps even earn a living out of SN, will need to run a 24/7 online lightning node, or else they would be missing out on zaps.
Not to mention that zaps failing would be annoying for the sender too.
Not everyone can afford to open a channel straight to SN just to guarantee that they can control the fee. And having fees introduces friction to zapping.
And I can't speak for others, but I'm very happy with my wallet and don't intend to change it for another. I'll just stop zapping (after I withdraw my sats while they are still sats).

Quoted from SN comment [#523435](https://stacker.news/items/523435/r/Radentor?commentId=523941)

# Rationale

- Sats zapping to hodl invoice [can causes force closure of user's channel](https://stacker.news/items/523435/r/Radentor?commentId=523583).
- Sats zapping can be forever removed with fee credit zapping, for several reasons 1) avoiding force close 2) regulatory issue 3) some self-custodial wallet does not have LN address (Bitkit, Blitz Electrum, Green, Phoenix) 4) each sats zap will carry LN routing fee & pathfinding time which can be daunting for newbs
- Condemning user to "use Blixt Bro" or "use Zeus Bro" or "use Mutiny Bro" will create centralization, more wallets option ≈ decentralization
- BOLT12 Offers which can give user native & static payment identifier won't likely come soon
- The idea of a Bitcoin platform having it's own non-Bitcoin token is not new. Bisq already has it's own token that is tradeable against real sats long time ago.

# Acknowledgement

- Personally, sats zapping & autowithdrawal is not an issue for me as i use Blixt with persistent mode 24/7 but even right now autowithdrawal is still broken and can be replaced with this proposal
- I can't say for sure the priority of this but what i can surely say is that any US Bitcoin company will have a difficult time going forward

**Comment by huumn** [comment]
Moving to discussion because this isn’t exactly a feature request. This is something that requires a lot of specification and consideration - legal and textual. 

We are working on this already. I explained it in detail in that post but it was misunderstood because of how complicated such a thing is.
